### ISSUE TITLE
Version insertion fails

### ISSUE THREAD

**Issue by ymartin59** [SynoCommunity/spkrepo#48]
A column on `version` table is too short for proper insertion. I bet for new DSM 6 columns `conf_privilege` and `conf_resource`.

I wonder if it makes sense to copy package configuration into database columns, moreover in Version entity, whereas it may be possible content differs for each architecture/build.

Trying to publish `dnscrypt-proxy_x64-6.1_2.0.44-5.spk` where `conf/privilege` file is 626 bytes large, here are server logs:
```
/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py:281: SAWarning: Object of type <Version> not in session, add operation along 'Package.versions' will not proceed
  util.warn(
[2020-07-26 06:31:09 +0000] [10] [ERROR] Error handling request /packages
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(255)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 134, in handle
    self.handle_request(listener, req, client, addr)
  File "/usr/local/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 175, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.8/site-packages/werkzeug/middleware/proxy_fix.py", line 169, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.8/site-packages/flask_restful/__init__.py", line 272, in error_router
    return original_handler(e)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.8/site-packages/flask/_compat.py", line 38, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/site-packages/flask_restful/__init__.py", line 272, in error_router
    return original_handler(e)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.8/site-packages/flask/_compat.py", line 38, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/src/app/spkrepo/views/api.py", line 47, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/flask_restful/__init__.py", line 468, in wrapper
    resp = resource(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/flask/views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/flask_restful/__init__.py", line 583, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/usr/src/app/spkrepo/views/api.py", line 282, in post
    db.session.commit()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/scoping.py", line 163, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 1042, in commit
    self.transaction.commit()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 504, in commit
    self._prepare_impl()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 483, in _prepare_impl
    self.session.flush()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2523, in flush
    self._flush(objects)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2664, in _flush
    transaction.rollback(_capture_exception=True)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.raise_(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 178, in raise_
    raise exception
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2624, in _flush
    flush_context.execute()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute
    rec.execute(self)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 586, in execute
    persistence.save_obj(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 239, in save_obj
    _emit_insert_statements(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 1135, in _emit_insert_statements
    result = cached_connections[connection].execute(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1014, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1127, in _execute_clauseelement
    ret = self._execute_context(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1317, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1511, in _handle_dbapi_exception
    util.raise_(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 178, in raise_
    raise exception
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.DataError: (psycopg2.errors.StringDataRightTruncation) value too long for type character varying(255)

[SQL: INSERT INTO version (package_id, version, upstream_version, changelog, report_url, distributor, distributor_url, maintainer, maintainer_url, dependencies, conf_dependencies, conflicts, conf_conflicts, conf_privilege, conf_resource, install_wizard, upgrade_wizard, startable, license, insert_date) VALUES (%(package_id)s, %(version)s, %(upstream_version)s, %(changelog)s, %(report_url)s, %(distributor)s, %(distributor_url)s, %(maintainer)s, %(maintainer_url)s, %(dependencies)s, %(conf_dependencies)s, %(conflicts)s, %(conf_conflicts)s, %(conf_privilege)s, %(conf_resource)s, %(install_wizard)s, %(upgrade_wizard)s, %(startable)s, %(license)s, now()) RETURNING version.id]
[parameters: {'package_id': 140, 'version': 5, 'upstream_version': '2.0.44', 'changelog': 'Update to 2.0.44', 'report_url': None, 'distributor': 'SynoCommunity', 'distributor_url': 'https://synocommunity.com/', 'maintainer': 'publicarray', 'maintainer_url': '', 'dependencies': None, 'conf_dependencies': None, 'conflicts': 'DNSServer', 'conf_conflicts': None, 'conf_privilege': '{\n    "defaults": {\n        "run-as": "package"\n    },\n    "username": "sc-dnscrypt-proxy",\n    "ctrl-script": [{\n        "action": "preinst",\ ... (359 characters truncated) ... ": "root"\n    }, {\n        "action": "start",\n        "run-as": "root"\n    }, {\n        "action": "stop",\n        "run-as": "root"\n    }]\n}\n', 'conf_resource': None, 'install_wizard': True, 'upgrade_wizard': False, 'startable': None, 'license': None}]
(Background on this error at: http://sqlalche.me/e/13/9h9h)
```

**Comment by ymartin59** [comment]
Have applied following statement on PostgreSQL database:
```
docker container exec -it -u postgres <CID> /usr/bin/psql
postgres=# ALTER TABLE version ALTER COLUMN conf_privilege TYPE varchar(2048);
ALTER TABLE
postgres=# ALTER TABLE version ALTER COLUMN conf_resource TYPE varchar(2048);
ALTER TABLE
```

**Comment by Diaoul** [comment]
Fixed with 899e5bdcab6ab601c375ec6c88b49542bfe49635
